src/main/java/org/Quotes/Main.java

```diff
@@ -0,0 +1,113 @@
+     */
+    public static void main(String[] args) {
+        try {
+            // Check if the file exists and is not empty
+            File file = new File(FILE_PATH);
+            if (file.exists() && file.length() > 0) {
+                // Read the quote of the day from the file
+                String json = new String(java.nio.file.Files.readAllBytes(Paths.get(FILE_PATH)));
+                Gson gson = new Gson();
+                Quote quote = gson.fromJson(json, Quote.class);
+                String fileDate = quote.contents.quotes[0].date;
+                String todayDate = LocalDate.now().format(DateTimeFormatter.ISO_DATE);
+                if (fileDate.equals(todayDate)) {
+                    System.out.println("From The .json File:");
+                    System.out.println("Today's (" + quote.contents.quotes[0].date + ") Quote:");
+                    System.out.println('"'+quote.contents.quotes[0].quote+'"');
+                    System.out.println("By : " + quote.contents.quotes[0].author);
+                }
+                else {
+                    System.out.println("Found: \n(" + quote.contents.quotes[0].date + ") Quote: ");
+                    System.out.print('"'+quote.contents.quotes[0].quote+'"');
+
+                    System.out.println("Updating the quote !!");
+                    // Delete the file
+                    file.delete();
+                    // Fetch the quote of the day from the API
+                    getResponseFromApi();
+                }
+            } else {
+                // Fetch the quote of the day from the API
+                getResponseFromApi();
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while reading or writing the file. Error message: " + e.getMessage());
+        } catch (JsonSyntaxException e) {
+            System.out.println("The API response is not in the expected format. Please try again later.");
+        } catch (Exception e) {
+            System.out.println("An error occurred. Error message: " + e.getMessage());
+        }
+    }
+    /**
+     * Method for saving the response of the API call to a JSON file
+     * @param request the request object for the API call
+     * @param client the OkHttpClient object for executing the API call
+     * @return the Quote object obtained from the API call
+     */
+    public static Quote saveResponse(Request request, OkHttpClient client) {
+        try (Response response = client.newCall(request).execute()) {
+            // Get the JSON response from the API
+            String json = response.body().string();
+            // Create a Gson object to parse the JSON
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            // Convert the JSON to a `Quote` object
+            Quote quote = gson.fromJson(json, Quote.class);
+
+            // Write the `Quote` object to a file
+            try (FileWriter writer = new FileWriter(FILE_PATH)) {
```

Comment:
This responsibility can be handled by a FileManager class or maybe some class with a meaningful name